File: backend/scripts/generate_qr.py
#!/usr/bin/env python3
import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from dotenv import load_dotenv
load_dotenv(os.path.join(os.path.dirname(__file__), "../.env"))

import qrcode
from pathlib import Path
from sqlmodel import Session, select

from app.database import engine
from app.models import WastePoint

APP_BASE_URL = os.getenv("APP_URL", "http://localhost:5173")
OUTPUT_DIR = Path(__file__).parent / "qr-codes"


def main():
    OUTPUT_DIR.mkdir(exist_ok=True)
    with Session(engine) as session:
        points = session.exec(select(WastePoint)).all()
        for p in points:
            # Frontend resident route is /r?b=<waste_point_id> (see App.jsx).
            url = f"{APP_BASE_URL}/r?b={p.id}"
            img = qrcode.make(url)
            out_path = OUTPUT_DIR / f"bin_{p.id}.png"
            img.save(str(out_path))
            print(f"✓ bin_{p.id}.png  →  {url}")
    print(f"\nGenerated {len(points)} QR codes in {OUTPUT_DIR}")


if __name__ == "__main__":
    main()

File: backend/scripts/generateQR.js
#!/usr/bin/env node
// QR Code Generator — Member A
// Generates one PNG per waste point: bin_<id>.png → ./qr-codes/
// Usage: npm run qr:generate

require('dotenv').config({ path: '../.env' });
const QRCode = require('qrcode');
const path = require('path');
const fs = require('fs');
const prisma = require('../prisma/client');

const APP_BASE_URL = process.env.APP_URL || 'http://localhost:5173';
const OUTPUT_DIR = path.join(__dirname, 'qr-codes');

async function main() {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
  const points = await prisma.wastePoint.findMany();

  for (const p of points) {
    const url = `${APP_BASE_URL}/report?bin=${p.id}&lat=${p.lat}&lng=${p.lng}`;
    const file = path.join(OUTPUT_DIR, `bin_${p.id}.png`);
    await QRCode.toFile(file, url, { width: 400 });
    console.log(`✓ bin_${p.id}.png  →  ${url}`);
  }

  console.log(`\nGenerated ${points.length} QR codes in ${OUTPUT_DIR}`);
}

main()
  .catch(e => { console.error(e); process.exit(1); })
  .finally(() => prisma.$disconnect());
